### PR TITLE
MBS-7879 improve documentation about dealing with flaky tests

### PR DESCRIPTION
**Pull request by int02h**

### CODE REVIEW

File: docs/content/menu/index.md [hunk 1]
```diff
@@ -70,5 +70,7 @@ headless: true
     - [Релиз]({{< ref "/docs/cd/Release.md" >}})
     - [Артефакты сборки]({{< ref "/docs/cd/Artifacts.md" >}})
     - [QApps]({{< ref "/docs/cd/QApps.md" >}})
+- Development
```
Comment (eugene-krivobokov) [inline docs/content/menu/index.md]: Testing?

File: docs/content/menu/index.md [hunk 2]
```diff
@@ -70,5 +70,7 @@ headless: true
...
+    - [Работа с флакующими тестами]({{< ref "docs/development/DealingWithFlaky.md" >}})
```
Comment (eugene-krivobokov) [inline docs/content/menu/index.md]: My proposal is to make one page - FlakyTests and merge into it all info about a process and `@Flaky` annotation.

File: docs/content/docs/development/DealingWithFlaky.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
+---
+title: Работа с флакующими тестами
+type: docs
+---
+
+# Работа с флакующими тестами
+
+## Что делать запрещается
```
Comment (eugene-krivobokov) [inline docs/content/docs/development/DealingWithFlaky.md]: Запрет без автоматизации ничем не отличается от рекомендации.
Предлагаю назвать соответственно: "Что не надо делать" и т.п.

А если что-то физически не могу сделать, то проще вообще не упоминать. Я же все равно это не могу сделать.

File: docs/content/docs/development/DealingWithFlaky.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+1. Игнорировать тесты в TeamCity. Вы просто "подложите свинью" своим коллегам в будущем.
+1. Озеленять билд через Mark As Successful. Мотивация та же, что и в п. 1.
+1. Помечать тест как `@Ignore`. Есть специальная аннотация [`@Flaky`]({{< ref "/docs/test/FlakyTests.md" >}}).
```
Comment (eugene-krivobokov) [inline docs/content/docs/development/DealingWithFlaky.md]: Нумерованные списки нужны только для описания последовательных шагов.
https://avito-tech.github.io/avito-android/docs/contributing/docs/#ordered-vs-unordered-lists

File: docs/content/docs/development/DealingWithFlaky.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+## Убедись что тест действительно флакует
+
+1. Посмотри в чаты [#android-dev](http://links.k.avito.ru/slackandroiddev) или 
+   [#android-autotesting](http://links.k.avito.ru/slackandroidautotesting). Вероятно, проблема уже известна.
+1. Посмотри [статистику стабильности теста]({{< ref "#где-посмотреть-статистику-по-стабильности-тестов" >}})
+1. Для проверки запускай тест в несколько прогонов в [динамической конфигурации]({{< ref "/docs/test/Run.md#custom-run" >}}).
```
Comment (eugene-krivobokov) [inline docs/content/docs/development/DealingWithFlaky.md]:
```suggestion
1. Запусти тест для проверки в несколько прогонов в [динамической конфигурации]({{< ref "/docs/test/Run.md#custom-run" >}}).
```

File: docs/content/menu/index.md (hunk 1 above)
Comment (int02h) [reply docs/content/menu/index.md]: We already have the Testing section. I wanted to make a new section where we will store documents about the development process. Not only testing.

Comment (eugene-krivobokov) [reply docs/content/menu/index.md]: Опиши пожалуйста весь концепт. Меняем общую структуру документации на ходу, замысел неочевиден.

Не понимаю почему работа с нестабильными тестами относится к разработке, но не к тестированию.

File: docs/content/menu/index.md (hunk 2 above)
Comment (int02h) [reply docs/content/menu/index.md]: `FlakyTests` is more about the technical part of `@Flaky` annotation. But I wanted to make documentation that describes the process.

Comment (int02h) [reply docs/content/menu/index.md]: Also, I think that mixing languages in one document is a bad idea.

File: docs/content/docs/development/DealingWithFlaky.md (hunk 3 above)
Comment (int02h) [reply docs/content/docs/development/DealingWithFlaky.md]: Поправил

File: docs/content/docs/development/DealingWithFlaky.md (hunk 4 above)
Comment (int02h) [reply docs/content/docs/development/DealingWithFlaky.md]: Поправил

File: docs/content/menu/index.md (hunk 2 above)
Comment (eugene-krivobokov) [reply docs/content/menu/index.md]: Agree

File: docs/content/menu/index.md (hunk 1 above)
Comment (int02h) [reply docs/content/menu/index.md]: Обсудили голосом. Структуру переделал.

**Review by eugene-krivobokov — APPROVED** [review]

File: docs/content/docs/test/FlakyTests.md [hunk 6]
```diff
@@ -1,53 +1,58 @@
...
+## Где посмотреть статистику по стабильности тестов?
+
+- [Общая статистика по тестам (internal)](http://links.k.avito.ru/FR)
+- [История нестабильности теста (internal)](http://links.k.avito.ru/5W)
+
+Обратите внимание на параметры фильтрации.
+
+## Когда убирать аннотацию `@Flaky`
+
+Если вы внесли изменения, которые чинят исходную причину флакования теста, то в том же PR
```
Comment (sergioboi) [inline docs/content/docs/test/FlakyTests.md:60]: Also, we could provide an option to look at BI statistic board

File: docs/content/docs/test/FlakyTests.md [hunk 7]
```diff
@@ -1,53 +1,58 @@
...
 
-## Suppress `@Flaky` test failures
+## Если не удалось найти причину
 
-To suppress failures of @Flaky tests just add `suppressFlaky = true` to your `uiTests` step.
-For more information on configuring build steps for the CI Steps plugin, please refer to [this document]({{< ref "/docs/projects/CISteps.md" >}}).
+1. Обратись к автору тесту или юниту, который за него отвечает.
+1. Если тест уже чинят, то можно дождаться фикса.
+1. Если проблема новая, то договоритесь о том, кто поставить аннотацию `@Flaky` на тест.
+1. В аннотации обязательно укажите причину, а лучше задачу, в которой тест будет исправлен.
```
Comment (sergioboi) [inline docs/content/docs/test/FlakyTests.md]: Add buildurl, reporturl to annotation

File: docs/content/docs/test/FlakyAnnotation.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+class MyAwesomeTests {
+    @Flaky
+    @Test
+    fun foobar() {
+        // ...
+    }
+}
+```
+
+## Suppress `@Flaky` test failures
```
Comment (sergioboi) [inline docs/content/docs/test/FlakyAnnotation.md:52]: It's not clear now how will change test behavior on CI after @Flaky will be added
We can describe current behavior that all Flaky don't run on PRs

Comment (int02h) [reply docs/content/docs/test/FlakyAnnotation.md:52]: This is described in `test/Run.md`. You can find all information about running tests in this document.

File: docs/content/docs/test/FlakyTests.md (hunk 7 above)
Comment (int02h) [reply docs/content/docs/test/FlakyTests.md]: Add this info.

File: docs/content/docs/test/FlakyTests.md (hunk 6 above)
Comment (int02h) [reply docs/content/docs/test/FlakyTests.md:60]: Added reference to previous section

Comment (sergioboi) [reply docs/content/docs/test/FlakyTests.md:60]: I only want that to become a more direct recommendation. Use BI when you want to remove `Flaky`

File: docs/content/docs/test/FlakyAnnotation.md (hunk 8 above)
Comment (sergioboi) [reply docs/content/docs/test/FlakyAnnotation.md:52]: I think when you give any advice to use something you need to declare Rules
You said that developers must use `Flaky` but didn't answer all question they would have
I prefer to see all answers or all links to answers in one place with usage algorithm

Comment (int02h) [reply docs/content/docs/test/FlakyAnnotation.md:52]: This doc was just renamed. In this PR I didn't have an intention to change it.

**Pull request merged**